frontend/src/scenes/billing/billingLogic.ts

```diff
@@ -66,7 +66,7 @@ export const billingLogic = kea<billingLogicType>({
     }),
     listeners: ({ values }) => ({
         subscribeSuccess: () => {
-            const { subscription_url } = values.billingSubscription
+            const { subscription_url } = values.billingSubscription || {}
```

Comment:
Hmm. In this case the `|| {}` is misleading right since we know `billingSubscription` should exist on subscribeSuccess.

Wouldn't using `values.billingSubscription!` here be better since we know (but typescript doesn't) it's not null?

Reply:
Hmm, you're right. The value should be set by the time we're here, but it's not obvious to TS.
The better option is to probably use the `{ billingSubscription }` argument in the listener (payload of the success action) and not take it from the value. I'll make a fix.

Reply:
I merged from master and added [this](https://github.com/PostHog/posthog/pull/2027/commits/dfdaf4f1467dc2f55d2cea4b6b7d5e569a45fe00).

```js
    listeners: () => ({
        subscribeSuccess: ({ billingSubscription }) => {
            if (billingSubscription?.subscription_url) {
                window.location.href = billingSubscription.subscription_url
            }
        },
    }),
```

Unfortunately the key `billingSubscription` for the `subscribeSuccess` action is also of type `BillingSubscription | null`, so just writing it out like this seems to make everyone happy.

I made an issue in kea-typegen to fix this: https://github.com/keajs/kea-typegen/issues/10

Reply:
Thanks for sharing the context! :)